overlord/state/taskrunner.go

```diff
@@ -143,7 +145,15 @@ func (r *TaskRunner) SetBlocked(pred func(t *Task, running []*Task) bool) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	r.blocked = pred
+	r.blocked = []blockedFunc{pred}
+}
+
+// AddBlocked adds a predicate function to decide whether to block a task from running based on the current running tasks. It can be used to control task serialisation. All added predicates are considered in turn until one returns true, or none.
+func (r *TaskRunner) AddBlocked(pred func(t *Task, running []*Task) bool) {
```

Comment:
Right, we will need this with a single taskrunner. I think though that `SetBlocked` should just implement 'append' semantics and should no longer set a single predicate, overwriting anything that was set already. I think the old semantics of SetBlocked could go away right now without waiting for other managers to be updated.

Reply:
are you suggesting to change SetBlocked to work like AddBlocked and not add AddBlocked?  name-wise is a bit strange. Or to just keep AddBlocked and remove SetBlocked?

Reply:
Agree about the name, AddBlocked is a better one. So yes, I'm suggesting to remove SetBlocked  (as it's kinda dangerous) and only have AddBlocked.

Reply:
the follow uses AddBlocked everywhere,  it would make more sense to me to do the removing of SetBlocked there I think without making this one touch more places

Reply:
OK, sounds good.